pkg/registry/csv.go

```diff
@@ -412,3 +414,61 @@ func (csv *ClusterServiceVersion) GetSubstitutesFor() string {
 	}
 	return substitutesFor
 }
+
+func (csv *ClusterServiceVersion) UnmarshalJSON(data []byte) error {
+
+	if err := csv.UnmarshalSpec(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalTypeMeta(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalObjectMeta(data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (csv *ClusterServiceVersion) UnmarshalSpec(data []byte) error {
+	var m map[string]json.RawMessage
+	if err := json.Unmarshal(data, &m); err != nil {
+		return errors.New(prettyunmarshaler.NewJSONUnmarshalError(data, err).Pretty())
+	}
+
+	for k, v := range m {
+		switch k {
+		case "spec":
+			var spec json.RawMessage
+			if err := json.Unmarshal(v, &spec); err != nil {
+				return errors.New(prettyunmarshaler.NewJSONUnmarshalError(v, err).Pretty())
+			}
+			csv.Spec = spec
+		}
+	}
```

Comment:
If we only care about the `spec` key is this loop necessary? Maybe something like
```go
specVal, ok := m["spec"]
if !ok {
  // seems like the current intention is to continue on if the spec field isn't found
  // but could also return an error here if we wanted
  return nil
}

var spec json.RawMessage
if err := json.Unmarshal(v, &spec); err != nil {
	return errors.New(prettyunmarshaler.NewJSONUnmarshalError(v, err).Pretty())
}
csv.Spec = spec
```
is better suited for this case?